src/openvic-simulation/country/CountryInstance.cpp

```diff
@@ -927,7 +932,110 @@ bool CountryInstance::update_rule_set() {
+	case COUNTRY_STATUS_SECONDARY_POWER:
+		modifier_sum.add_modifier_nullcheck(static_modifier_cache.get_secondary_power(), country_source);
+		break;
+	case COUNTRY_STATUS_CIVILISED:
+		modifier_sum.add_modifier_nullcheck(static_modifier_cache.get_civilised(), country_source);
+		break;
+	default:
+		modifier_sum.add_modifier_nullcheck(static_modifier_cache.get_uncivilised(), country_source);
+	}
+	if (is_disarmed()) {
+		modifier_sum.add_modifier_nullcheck(static_modifier_cache.get_disarming(), country_source);
+	}
+	modifier_sum.add_modifier_nullcheck(static_modifier_cache.get_war_exhaustion(), country_source, war_exhaustion);
+	modifier_sum.add_modifier_nullcheck(static_modifier_cache.get_infamy(), country_source, infamy);
+	modifier_sum.add_modifier_nullcheck(static_modifier_cache.get_literacy(), country_source, national_literacy);
+	modifier_sum.add_modifier_nullcheck(static_modifier_cache.get_plurality(), country_source, plurality);
+	// TODO - difficulty modifiers, war, peace, debt_default_to, bad_debter, generalised_debt_default,
+	//        total_occupation, total_blockaded, in_bankrupcy
+
+	// TODO - handle triggered modifiers
+
+	if (ruling_party != nullptr) {
+		for (Issue const* issue : ruling_party->get_policies()) {
+			// The ruling party's issues here could be null as they're stored in an IndexedMap which has
+			// values for every IssueGroup regardless of whether or not they have a policy set.
+			modifier_sum.add_modifier_nullcheck(issue, country_source);
+		}
+	}
+
+	for (Reform const* reform : reforms) {
+		// The country's reforms here could be null as they're stored in an IndexedMap which has
+		// values for every ReformGroup regardless of whether or not they have a reform set.
+		modifier_sum.add_modifier_nullcheck(reform, country_source);
+	}
+
+	modifier_sum.add_modifier_nullcheck(national_value, country_source);
+
+	modifier_sum.add_modifier_nullcheck(tech_school, country_source);
+
+	for (Technology const& technology : *technology_unlock_levels.get_keys()) {
+		if (is_technology_unlocked(technology)) {
+			modifier_sum.add_modifier(technology, country_source);
+		}
+	}
+
+	for (Invention const& invention : *invention_unlock_levels.get_keys()) {
+		if (is_invention_unlocked(invention)) {
+			modifier_sum.add_modifier(invention, country_source);
+		}
+	}
+
+	// Add province base modifiers (with local province modifier effects removed)
+	using enum ModifierEffect::target_t;
+	static constexpr ModifierEffect::target_t NOT_PROVINCE = ALL_TARGETS & ~PROVINCE;
+
+	for (ProvinceInstance const* province : owned_provinces) {
+		modifier_sum.add_modifier_sum_filter_targets(province->get_modifier_sum(), NOT_PROVINCE);
+	}
+
+	// This has to be done after adding all province modifiers to the country's sum, otherwise provinces earlier in the list
+	// wouldn't be affected by modifiers from provinces later in the list.
+	for (ProvinceInstance* province : owned_provinces) {
+		province->contribute_country_modifier_sum(modifier_sum);
```

Comment:
This still feels misplaced.

I prefer the following

1. loop through provinces to calculate their modifier sum (using province modifiers and such)
2. calculate country modifier sum by looping through all owned (and controlled?) provinces
3. perform daily updates such as pop growth and production by summing the province modifier sum + country modifier sum

1. is currently implemented in the map instance loop in this PR.
2. is currently implemented in the country instance in this PR (line 1015)

That just leaves this code to be moved to the province responsibility.
As we discussed on Discord, you want to try approaches, so I'll approve anyway.
This comment is mainly for clarity and documentation.